packages/example/src/pages/guides/v4-migration.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: V4.0 Migration
+description: Guide for migrating to v4.0 of the theme
+---
+
+<PageDescription>
+
+Carbon's Gatsby Theme uses version 5 of the Gatsby library. This update combines
+the control and scalability of server-side rendering with the performance of
+static-site generation. The major version change comes with the potential for
+breaking changes, this guide aims to cover the most likely issues a site built
+with the Carbon Gatsby theme might encounter. To have the smoothest possible
+experience, we recommend [upgrading to v3](/guides/v3-migration) of Carbon's
+Gatsby Theme first if you haven't already done so.
+
+</PageDescription>
+
+## Breaking changes
+
+1. Bumped minimum version of Gatsby to `v5`
+1. Gatsby files (i.e. gatsby-config.js and gatsby-node.js), need to be converted
+   to **.mjs** (node modules) files.
+1. TBD...
+
+## Gatsby migration guide
+
+In this section we'll cover the most likely issues site's built with the Carbon
+Gatsby theme might encounter when migrating. If you encounter any issues, or
+your site leverages more advanced Gatsby configuration, check to see if it's
+covered in the
+[official Gatsby v5 migration guide](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v4-to-v5/).
+To facilitate an easy upgrade path, we recommend
+[upgrading to v3](/guides/v3-migration) of Carbon's Gatsby Theme first if you
+haven't already done so.
+
+<InlineNotification>
+
+Note: We'll use yarn for the shell commands here. Replace them with the `npm`
+equivalent if your project uses `npm` as its package manager.
+
+</InlineNotification>
+
+1. Update Gatsby to v5
+
+   ```shell
+   yarn add gatsby@latest-v5
+   ```
+
+1. Update Gatsby plugins (including the theme)
+
+   ```shell
+   yarn upgrade-interactive --latest
+   ```
+
+1. Update graphql queries using the codemod:
+
+   ```shell
+   npx gatsby-codemods@latest sort-and-aggr-graphql
+   ```
+
+1. Homepage and default layouts have been moved from **templates** to **layout**
+   components. Update any shadowing or links to template layouts.
```

Comment:
Should we move this under breaking changes?

```jsx
// Before
import Layout from 'gatsby-theme-carbon/src/templates/Default';
// After
import Layout from 'gatsby-theme-carbon/src/components/Layouts/Default';
```

If you are shadowing the homepage you need to update the file path
`packages/example/src/components/Homepage.js`

```jsx
// Before
import HomepageTemplate from 'gatsby-theme-carbon/src/templates/Homepage';
// After
import HomepageTemplate from 'gatsby-theme-carbon/src/components/Layouts/Homepage';
```